### PR TITLE
spring-cloud-gateway-configuration

### PR DESCRIPTION
**Pull request by jah3**

### CODE REVIEW

File: gateway-notification-service/src/main/java/md/maib/gatewaynotificationservice/config/RoutesConfig.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.servlet.function.RequestPredicates;
+import org.springframework.web.servlet.function.RouterFunction;
+import org.springframework.web.servlet.function.ServerResponse;
+
+@Configuration
+public class RoutesConfig {
+
+    @Bean
+    public RouterFunction<ServerResponse>  notificationRouteService(){
```
Comment (taragrammatonn) [inline gateway-notification-service/src/main/java/md/maib/gatewaynotificationservice/config/RoutesConfig.java:15]: you can also add this in application yml file, but still ok

**Review by taragrammatonn — APPROVED** [review]

**Comment by taragrammatonn** [comment]
add some test

File: gateway-notification-service/src/test/java/md/maib/gatewaynotificationservice/config/RoutesConfigTest.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    @Autowired
+    private ApplicationContext applicationContext;
+
+    @Test
+    public void testNotificationRouteServiceBeanCreated() {
+        // Retrieve the bean from the application context
+        RouterFunction<ServerResponse> bean = applicationContext.getBean("notificationRouteService", RouterFunction.class);
+
+        // Assert that the bean is not null
+        assertThat(bean).isNotNull();
```
Comment (taragrammatonn) [inline gateway-notification-service/src/test/java/md/maib/gatewaynotificationservice/config/RoutesConfigTest.java:24]: It's a just null check, need to test if your request pass getaway and return some valid result

**Review by taragrammatonn — CHANGES_REQUESTED** [review]
Pls add some request example with response assert

File: gateway-notification-service/pom.xml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        <spring-cloud.version>2023.0.3</spring-cloud.version>
+    </properties>
+    <dependencies>
+        <dependency>
+            <groupId>org.springframework.cloud</groupId>
+            <artifactId>spring-cloud-starter-gateway-mvc</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-test</artifactId>
```
Comment (taragrammatonn) [inline gateway-notification-service/pom.xml:40]: Redundant is case of starter test

File: gateway-notification-service/pom.xml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-test</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-test</artifactId>
```
Comment (taragrammatonn) [inline gateway-notification-service/pom.xml:50]: Twice the same dependency

**Review by taragrammatonn — APPROVED** [review]
Fix comments before merge

File: gateway-notification-service/pom.xml (hunk 4 above)
Comment (jah3) [reply gateway-notification-service/pom.xml:50]: Sure 👍🏽

**Pull request merged**